src/cache-manager.js

```diff
@@ -50,22 +50,39 @@ class CacheManager {
     assert(rawUrl);
     this.debug(`putting ${rawUrl}`);
 
+    let mutex = new Mutex(this.redis, this.lockKey(rawUrl));
```

Comment:
I oppose this locking... It can't possibly explain anything.

I see NO possibility that currently copying could possibly in any way cause an issue.
S3 docs are pretty clear about that.

So let's not do this... Locking when not necessary is a recipe for disaster.